flexmeasures/cli/utils.py

```diff
@@ -328,17 +328,17 @@ def validate_url(ctx, param, value):
     :param value:   The URL to validate.
     """
     url_regex = re.compile(
-        r"^(https?|ftp)://"
-        r"(([A-Za-z0-9$_.+!*\'(),;?&=-]|%[0-9A-Fa-f]{2})+@)?"  # User and pass
+        r"^(https?|ftp)://"  # Protocol: http, https, or ftp
         r"(([A-Za-z0-9.-]+)\.([A-Za-z]{2,6})|"  # Domain name
```

Comment:
I believe you removed too much from this regex - why not more than just the domain?

Also there seems to be a regex error when I create a new account, so it seems you have not really tested if this works with a real URL. (I used https://smartdodos.com/wp-content/uploads/2021/12/logo.webp)

Here is my error:

```
± flexmeasures add account --name LogoDummy --logo-url https://smartdodos.com/wp-content/uploads/2021/12/logo.webp
(...)
Traceback (most recent call last):
  File "/home/nicolas/envs/fm-3.10/bin/flexmeasures", line 33, in <module>
    sys.exit(load_entry_point('flexmeasures', 'console_scripts', 'flexmeasures')())
  File "/home/nicolas/envs/fm-3.10/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/nicolas/envs/fm-3.10/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/nicolas/envs/fm-3.10/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/nicolas/envs/fm-3.10/lib/python3.10/site-packages/click/core.py", line 1686, in invoke
    sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)
  File "/home/nicolas/envs/fm-3.10/lib/python3.10/site-packages/click/core.py", line 943, in make_context
    self.parse_args(ctx, args)
  File "/home/nicolas/envs/fm-3.10/lib/python3.10/site-packages/click/core.py", line 1408, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "/home/nicolas/envs/fm-3.10/lib/python3.10/site-packages/click/core.py", line 2400, in handle_parse_result
    value = self.process_value(ctx, value)
  File "/home/nicolas/envs/fm-3.10/lib/python3.10/site-packages/click/core.py", line 2362, in process_value
    value = self.callback(ctx, self, value)
  File "/home/nicolas/workspace/seita/flexmeasures/flexmeasures/cli/utils.py", line 330, in validate_url
    url_regex = re.compile(
  File "/usr/lib/python3.10/re.py", line 251, in compile
    return _compile(pattern, flags)
  File "/usr/lib/python3.10/re.py", line 303, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.10/sre_compile.py", line 788, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.10/sre_parse.py", line 955, in parse
    p = _parse_sub(source, state, flags & SRE_FLAG_VERBOSE, 0)
  File "/usr/lib/python3.10/sre_parse.py", line 444, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
  File "/usr/lib/python3.10/sre_parse.py", line 843, in _parse
    raise source.error("missing ), unterminated subpattern",
re.error: missing ), unterminated subpattern at position 16
```

Reply:
Apologies for the delay in responding to this comment.

I left the regex patterns in place, as they handle the main validation (protocol and domain).

I'll investigate the error further to resolve it and push an update

Reply:
Apart from the error, I believe only protocol and domain will miss out the path that all URLs have. Some URLs might have a port as well - I'm unsure why you threw out most of it. Felix only suggested to add a link to where we got it from.

Reply:
After further investigation, I discovered that the issue was with the regex validation. The additional check for IP addresses, which I had removed, was actually necessary to ensure the overall validation worked correctly. 

Initially, I believed that validating just the domain and protocol would be sufficient, but I underestimated the complexity of the regex. Moving forward, it might be more reliable to use a dedicated package for this validation, but I believe for now this should suffice.

Reply:
I did consider URLs with ports but assumed that users would primarily provide standard image links with typical domains. However, if you think there's a chance our users might need to include URLs with ports, I'm happy to add that validation back in.

I didn't include the other patterns because I wasn't entirely clear on how they worked—especially the one related to fragments. The path and query string validations also seemed more like nice-to-have features rather than essentials.

Reply:
Query string might not happen a lot in our URL case here, but the path is essential (image file name would be part of it, see my example). Maybe a URL could even have a query string like `?file=logo.jpg` so if the regex can offer that, I'd keep it as well.

Reply:
I see, I’ve included the pattern for the path as well. I'll also keep the query string validation to cover cases like ?file=logo.jpg.